pulsar-broker/src/test/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentStickyKeyDispatcherMultipleConsumersTest.java

```diff
@@ -94,6 +94,15 @@ public void setup() throws Exception {
             new HashRangeAutoSplitStickyKeyConsumerSelector());
     }
 
+    @Test(timeOut = 10000)
+    public void testAddConsumerWhenClosed() throws Exception {
+        nonpersistentDispatcher.close().get();
+        Consumer consumer = mock(Consumer.class);
+        nonpersistentDispatcher.addConsumer(consumer);
+        verify(consumer, times(1)).disconnect();
+        assertEquals(0, nonpersistentDispatcher.getConsumers().size());
```

Comment:
Should we add a verify: `this consumer does not exist in the selector`?

Reply:
OK. I will update it later

Reply:
I find that it's not easy to access the `selector`, and verify `assertEquals(0, nonpersistentDispatcher.getConsumers().size());` is enough. So maybe we needn't change it @poorbarcode

Reply:
done